### PR TITLE
feat(Avatar): expose onError

### PR DESCRIPTION
**Pull request by paranoidjk**
## Example

```js
class Foo extends React.Component {
  state = {
   src: "https://cdn.com/foo.png@100h.src", // `@100h.src` is a cdn scale param, but it may fail 
 }
  handleImgError = () => {
    this.setState({
      src: "https://cdn.com//foo.png", // fallback to original big image
   });
 }
  render() {
     return <Avatar src={this.state.src} onError={this.handleImgError}/>
  }
}
```

First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [ ] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [ ] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [ ] Rebase before creating a PR to keep commit history clear.
* [ ] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

### CODE REVIEW

File: components/avatar/index.zh-CN.md [hunk 1]
```diff
@@ -15,3 +15,4 @@ title: Avatar
 | shape | 指定头像的形状 | Enum{ 'circle', 'square' } | `circle` |
 | size | 设置头像的大小 | Enum{ 'large', 'small', 'default' } | `default` |
 | src | 图片类头像的资源地址 | string | - |
+| onError | 图片加载失败的事件，返回 true 会关闭组件默认的 fallback 行为 | () => boolean | - |
```
Comment (yesmeck) [inline components/avatar/index.zh-CN.md]: 应该是返回 false？

Comment (paranoidjk) [reply components/avatar/index.zh-CN.md]: 默认不 retun 是 undefined，即 false，我感觉 true 明确一点，用户必须手动返回才能阻止。

还是说考虑监听了 onError 的都不需要组件的默认 fallback ？那改成 false 也行

Comment (yesmeck) [reply components/avatar/index.zh-CN.md]: 一般都是 false 是阻止默认行为啊，用 `===` 判断。

Comment (paranoidjk) [reply components/avatar/index.zh-CN.md]: 改了

**Comment by yesmeck** [comment]
3.7

**Review by yesmeck — APPROVED** [review]

**Comment by paranoidjk** [comment]
分支改好了

**Comment by afc163** [comment]
补充个用例吧

**Comment by paranoidjk** [comment]
加了用例，这个没法 reopen 迁移到 https://github.com/ant-design/ant-design/pull/11285

**Pull request closed without merging**